tyk-pro/templates/deployment-gw-repset.yaml

```diff
@@ -30,6 +30,13 @@ spec:
       release: {{ .Release.Name }}
   template:
     metadata:
+      {{- if .Values.enableIstioIngress }}
+      annotations:
+        traffic.sidecar.istio.io/excludeInboundPorts: "{{ .Values.gateway.containerPort}}"
+        traffic.sidecar.istio.io/includeInboundPorts: ""
+        traffic.sidecar.istio.io/excludeOutboundPorts: "{{ .Values.dash.containerPort }}"
+        traffic.sidecar.istio.io/includeOutboundPorts: ""
```

Comment:
Things are a bit unclear for me here: By definition `excludeInboundPorts` is: `A comma separated list of inbound ports to be excluded from redirection to Envoy. Only applies when all inbound traffic (i.e. '*') is being redirected.`, but then we set all inbound traffic to be disabled by giving it an empty list eg. `includeInboundPorts: ""`, so shouldn't it the wildcard character '*' be used here to configure redirection for all inbound ports?

Same applies for outbound ports. Here is docs for reference: https://istio.io/latest/docs/reference/config/annotations/

Reply:
It a bug in the istio implementation which means the defined excludeInboundPorts wont work unless you also provide the other annotation even if it's empty and does nothing.

Reply:
I'll have a play and see if wildcard affects things - although i'm comfortable leaving the possibility of all outward facing ports being available because that makes opening up TCP/gRPC apis a lot simpler